plugins/ansilexer.py

```diff
@@ -28,83 +29,288 @@
+            command = parameters.pop(0)
+            if command == 0:
+                self._bright = False
+                self._foreground = 'Default'
+                self._background = 'Default'
+            elif command == 1:
+                self._bright = True
+            elif command == 22:
+                self._bright = False
+            elif command >= 30 and command <= 37:
+                self._foreground = self._named_colors[command - 30]
+            elif command == 38:
+                mode = parameters.pop(0)
+                if mode == 2:
+                    rgb = self._read_rgb(parameters)
+                else:
+                    offset = parameters.pop(0)
+                    rgb = self._make_color_from_palette(offset)
+                self._foreground = 'ForegroundColor' + rgb
+            elif command == 39:
+                self._foreground = 'Default'
+            elif command >= 40 and command <= 47:
+                self._background = self._named_colors[command - 40]
+            elif command == 48:
+                mode = parameters.pop(0)
+                if mode == 2:
+                    rgb = self._read_rgb(parameters)
+                else:
+                    offset = parameters.pop()
+                    rgb = self._make_color_from_palette(offset)
+                self._background = rgb
+            elif command == 49:
+                self._background = 'Default'
+            elif command >= 90 and command <= 97:
+                self._foreground = ('Bright' +
+                                    self._named_colors[command - 90])
+            elif command >= 100 and command <= 107:
+                self._background = ('Bright' +
+                                    self._named_colors[command - 100])
+
+        if (self._bright
+            and not self._foreground.startswith('Bright')
+            and not self._foreground.startswith('ForegroundColor')):
+            token = 'Generic.AnsiBright' + self._foreground
         else:
+            token = 'Generic.Ansi' + self._foreground
+
+        if (self._background != 'Default'):
+            token += 'BackgroundColor' + self._background
+
+        if token == 'Generic.AnsiDefault':
             yield (match.start(), Text, text)
-            return
-
-        if color == '30':
-            token += 'Black'
-        elif color == '31':
-            token += 'Red'
-        elif color == '32':
-            token += 'Green'
-        elif color == '33':
-            token += 'Yellow'
-        elif color == '34':
-            token += 'Blue'
-        elif color == '35':
-            token += 'Magenta'
-        elif color == '36':
-            token += 'Cyan'
-        elif color == '37':
-            token += 'White'
-        elif color == '39':
-            token += 'Default'
         else:
-            yield (match.start(), Text, text)
-            return
+            yield (match.start(), string_to_tokentype(token), text)
+        return
```

Comment:
redundant return

Reply:
How embarrassing. Amended and pushed. Original branch renamed to ansilexer-pre-amend for your comparing convenience.